frontend/src/metabase/visualizations/lib/renderer_utils.js

```diff
@@ -94,6 +94,72 @@ export function reduceGroup(group, key, warnUnaggregated) {
   );
 }
 
+export const parseXValue = _.memoize(
+  (xValue, { isNumeric, isTimeseries, isQuantitative, unit }, warn) =>
+    // don't parse as timestamp if we're going to display as a quantitative scale, e.x. years and Unix timestamps
+    isTimeseries && !isQuantitative
+      ? HACK_parseTimestamp(xValue, unit, warn)
+      : isNumeric
+      ? xValue
+      : String(xValue),
+  // create cache key from args
+  // we need typeof so "2" and 2 don't have the same cache key
+  (x, options) => [x, typeof x, ...Object.values(options)].join(),
```

Comment:
Just to check, did you profile `parseXValue` before and after adding memoization?

Also, I think this is incorrect since `warn` is not included in the hash function. It looks like we call `getDatas` before `getXValues` via `getXAxisProps` so it's not a problem on the first render, but what about subsequent renders with the same values?